### PR TITLE
perf(tooltip): improve the performance of tooltip.

### PR DESCRIPTION
**Pull request by plainheart**
## Brief Information

This pull request is in the type of:

- [ ] bug fixing
- [ ] new feature
- [x] others

### What does this PR do?

As mentioned in #13680, the tooltip looks delayed and has an obvious lag. This brings the users a bad interaction experience.
This PR is attempting to improve the performance of the tooltip. Any idea is welcome.

### Fixed issues

- #13680
- #14204

### Before: What was the problem?

The moving of the tooltip has an obvious lag.

### After: How is it fixed in this PR?

https://user-images.githubusercontent.com/26999792/107198365-f6b50e00-6a2f-11eb-9b37-3081c71ca1ed.mp4

### Are there any API changes?

- [ ] The API has been changed.

### Related test cases or examples to use the new APIs

Please refer to `test/new-tooltip.html` or any other test cases related to `tooltip`.

### Merging options

- [ ] Please squash the commits into a single one when merge.

### Other information

Related PR in zrender: ecomfe/zrender#721

### CODE REVIEW

**Comment by pissang** [comment]
Wow looks really nice

**Review by 100pah — CHANGES_REQUESTED** [review]

**Comment by 100pah** [comment]
Wonderful job.

File: src/component/tooltip/helper.ts [hunk 1]
```diff
@@ -27,3 +30,37 @@ export function shouldTooltipConfine(tooltipModel: Model<TooltipOption>): boolea
         // In richText mode, the outside part can not be visible.
         : tooltipModel.get('renderMode') === 'richText';
 }
+
+function testStyle(styleProps: string[]): string | undefined {
+	const style = document.documentElement.style;
```
[the remark this answers is not in the document]
Comment (plainheart) [reply src/component/tooltip/helper.ts]: I used zrender `env.domSupported` to detect it.

File: src/component/tooltip/helper.ts [hunk 2]
```diff
@@ -27,3 +31,40 @@ export function shouldTooltipConfine(tooltipModel: Model<TooltipOption>): boolea
         // In richText mode, the outside part can not be visible.
         : tooltipModel.get('renderMode') === 'richText';
 }
+
+function testStyle(styleProps: string[]): string | undefined {
+    if (!env.domSupported) {
+        return;
```
Comment (100pah) [inline src/component/tooltip/helper.ts:37]: I am not sure that if here `undefined` returned, 
then `TRANSFORM_VENDOR` and `TRANSITION_VENDOR` will become `undefined`, 
then 
```js
const CSS_TRANSITION_VENDOR = toCSSVendorPrefix(TRANSITION_VENDOR, 'transition');
const CSS_TRANSFORM_VENDOR = toCSSVendorPrefix(TRANSFORM_VENDOR, 'transform');
```
An `undefined` will input to `toCSSVendorPrefix`.
And then a null pointer error will be thrown in `toCSSVendorPrefix`?

**Review by 100pah — CHANGES_REQUESTED** [review]

File: src/component/tooltip/helper.ts (hunk 2 above)
Comment (plainheart) [reply src/component/tooltip/helper.ts:37]: Sorry. You are right. I missed it.

**Review by 100pah — APPROVED** [review]

**Pull request merged**